src/main/java/com/thealgorithms/strings/ReverseWordInString.java

```diff
@@ -0,0 +1,41 @@
+package com.thealgorithms.strings;
+
+/**
+ Given an input string s, reverse the order of the words.
+ A word is defined as a sequence of non-space characters.
+ The words in s will be separated by at least one space.
+ Return a string of the words in reverse order concatenated by a single space.
+ Note that s may contain leading or trailing spaces or multiple spaces between two words.
+ The returned string should only have a single space separating the words.Do not include any extra spaces
+ */
+
+public class ReverseWordInString {
```

Comment:
This could be a _proper utility class_.
```suggestion
public final class ReverseWordInString {
    private ReverseWordInString() {
    }
```

Reply:
yes made the current class final but I need a public method there to access that method from other classes,
I think I am not able to understand this suggestion

Reply:
yes! Your class will be public. but you have to add a constructor.
The presence of this private constructor indicates that the class is designed to be a utility class, and its constructor is made private to prevent instances of this class from being created. This is a common practice in Java for utility classes that contain only static methods and don't need to be instantiated

Reply:
got your point, done the changes